72531-NEW Delete/Redistribute Install with Native Helm Chart (#1042)
replicatedhq/replicated-docs: docs/vendor/helm-overview.md

import TemplateLimitation from "../partials/helm/_helm-template-limitation.mdx"
import HelmCLILimitations from "../partials/helm/_helm-cli-limitations.mdx"

# About Deploying with Helm

Helm is a popular package manager for Kubernetes applications. If your application is already packaged using Helm, you can use Replicated to more easily distribute and manage your application. Using Replicated to distribute applications packaged with Helm provides additional functionality not available through Helm, such as preflight checks, support bundles, a user interface for collecting user configuration values, support for using private images, and more.

Replicated supports delivering an enterprise application as Helm charts, or including Helm charts as components of an application. An application can use more than one Helm chart, and can use more than a single instance of any Helm chart.

> To deploy an application with Helm, start by adding an existing Helm chart to a release in the Replicated vendor portal. For information about how to create a new release from an existing Helm chart, see [Creating Releases with Helm Charts](helm-release).

Reviewer: WDYT? "deploy an application" doesn't sound right since the chart is just part of the application.
```suggestion
To deploy a Helm chart, start by adding the Helm chart to a release in the Replicated vendor portal. For information about how to create a new release from an existing Helm chart, see [Creating Releases with Helm Charts](helm-release).
```

Users can install an application packaged with Helm charts using the app manager on an existing cluster or on a cluster provisioned by the Kubernetes installer. When installing with the app manager, users can either use the Replicated admin console UI or the kots CLI.

> The app manager installs applications packaged with Helm charts using either the **Native Helm** or **Replicated Helm** deployment method. You specify the deployment method in the Replicated HelmChart custom resource manifest file with the `useHelmInstall` flag. For more information, see [useHelmInstall](/reference/custom-resource-helmchart#usehelminstall) in _HelmChart_.

Reviewer: Same thoughts as above.
```suggestion
The app manager installs Helm charts using either the **Native Helm** or **Replicated Helm** deployment method. You specify the deployment method in the Replicated HelmChart custom resource manifest file with the `useHelmInstall` flag. For more information, see [useHelmInstall](/reference/custom-resource-helmchart#usehelminstall) in _HelmChart_.
```

The following describes the Native Helm and Replicated Helm deployment methods:

* **Native Helm (Recommended)**: The app manager uses the Helm binary to install and manage the lifecycle of the chart resources that are part of the application. This is the preferred method because it supports more features of Helm, such as hooks and weights.

  For more information, see [Native Helm](helm-processing#native-helm) in _How the App Manager Processes Helm Charts_. See also [Creating Releases with Helm Charts](helm-release).

  :::note

>   Migrating existing installations to the native Helm workflow is not supported. However, new Helm charts within an existing application can use the native Helm workflow and the features that come with it.

Reviewer: I'm just toying around, but this makes a bit more sense to me.
```suggestion
  Charts installed with Replicated Helm cannot be migrated to use Native Helm. However, Native Helm can always be used for charts that are newly added to a release, even if other charts in the release use Replicated Helm.
```

There are different limitations depending on if your customers install and manage the application with the app manager, or if they use the helm CLI directly:

* [Replicated Helm and Native Helm Limitations](#replicated-helm-limitations)
* [helm CLI Install Limitations](#helm-cli-limitations)

> ### Native Helm and Replicated Helm Limitations {#replicated-helm-limitations}

Author: Thanks!
